trainers/base_trainer.py

```diff
@@ -16,6 +16,39 @@
 )
 from trainers.evaluator import train_eval
 
+import numpy as np
+from itertools import islice
+from torch.nn.parallel import DistributedDataParallel as DDP 
+from torch.utils.data.distributed import DistributedSampler
+from trainers.utils import aggregate_value
+
+class CustomDataset:
```

Comment:
Pointless having this here. Make our dataclass subclass the torch dataloader. Then replace the get_batch  method with this get_item method

Reply:
* make our dataloader subclass/inherit the torch dataset